### PR TITLE
Mark error as nullable in delegates

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Mark error as nullable in delegates. The wrong attribute can lead to a crash if it's marked as force unwrapped in Swift.

## Related PRs or issues

[AB#86006](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/86006)

### CODE REVIEW

File: AppCenter/AppCenter/Internals/Channel/MSACChannelGroupDefault.m [hunk 1]
```diff
@@ -144,7 +144,7 @@ - (void)channel:(id<MSACChannelProtocol>)channel didSetEnabled:(BOOL)isEnabled a
                             }];
 }
 
-- (void)channel:(id<MSACChannelProtocol>)channel didFailSendingLog:(id<MSACLog>)log withError:(NSError *)error {
+- (void)channel:(id<MSACChannelProtocol>)channel didFailSendingLog:(id<MSACLog>)log withError:(nullable NSError *)error {
```
Comment (DmitriyKirakosyan) [inline AppCenter/AppCenter/Internals/Channel/MSACChannelGroupDefault.m:147]: Do we need it in `.m` file?

File: SasquatchMac/SasquatchMacSwift/AppDelegate.swift [hunk 2]
```diff
@@ -140,32 +140,32 @@ class AppDelegate: NSObject, NSApplicationDelegate, CrashesDelegate, CLLocationM
...
 
-  func crashes(_ crashes: Crashes, didSucceedSending errorReport: ErrorReport) {
+  func crashes(_ crashes: Crashes!, didSucceedSending errorReport: ErrorReport) {
     if errorReport.exceptionReason != nil {
       NSLog("Did succeed error report sending with: %@", errorReport.exceptionReason);
     }
   }
 
-  func crashes(_ crashes: Crashes, didFailSending errorReport: ErrorReport, withError error: Error) {
+  func crashes(_ crashes: Crashes!, didFailSending errorReport: ErrorReport!, withError error: Error) {
```
Comment (DmitriyKirakosyan) [inline SasquatchMac/SasquatchMacSwift/AppDelegate.swift:162]: shouldn't be `Error?` here?

File: AppCenter/AppCenter/Internals/Channel/MSACChannelGroupDefault.m (hunk 1 above)
Comment (MatkovIvan) [reply AppCenter/AppCenter/Internals/Channel/MSACChannelGroupDefault.m:147]: It's all optional, but for clarity/readability - yes

File: SasquatchMac/SasquatchMacSwift/AppDelegate.swift (hunk 2 above)
Comment (MatkovIvan) [reply SasquatchMac/SasquatchMacSwift/AppDelegate.swift:162]: It can be, but should not

File: SasquatchMac/SasquatchMacSwift/AppDelegate.swift [hunk 3]
```diff
@@ -140,32 +140,32 @@ class AppDelegate: NSObject, NSApplicationDelegate, CrashesDelegate, CLLocationM
...
 
-  func crashes(_ crashes: Crashes, willSend errorReport: ErrorReport) {
+  func crashes(_ crashes: Crashes!, willSend errorReport: ErrorReport!) {
     if errorReport.exceptionReason != nil {
       NSLog("Will send error report with: %@", errorReport.exceptionReason);
     }
   }
 
-  func crashes(_ crashes: Crashes, didSucceedSending errorReport: ErrorReport) {
+  func crashes(_ crashes: Crashes!, didSucceedSending errorReport: ErrorReport) {
```
Comment (DmitriyKirakosyan) [inline SasquatchMac/SasquatchMacSwift/AppDelegate.swift:156]:
```suggestion
  func crashes(_ crashes: Crashes!, didSucceedSending errorReport: ErrorReport!) {
```

**Review by russelarms — APPROVED** [review]

**Pull request merged**